src/ZoneServer/World/Maps/Map.cs

```diff
@@ -43,6 +41,13 @@ public class Map : IUpdateable
 		/// </summary>
 		private readonly Dictionary<int, Character> _characters = new();
 
+		/// <summary>
+		/// Collection of dummy characters.
+		/// <para>Key: <see cref="Character.Handle"/></para>
+		/// <para>Value: <see cref="Character"/></para>
+		/// </summary>
+		private readonly Dictionary<int, Character> _dummies = new Dictionary<int, Character>();
```

Comment:
Some of Melia methods takes in consideration that a character on the _characters dictionary have a connection and dummies don't have it. That should be the main difference. Well, I am out of ideas on how to improve it. I don't think it's should be overcomplicated: is it a dummy? Instead of adding to the _characters dictionary add it to the _dummies. 

Broadcasts is an example that we should be threating characters differently from dummies:

```
		/// <summary>
		/// Broadcasts packet to all characters on map.
		/// </summary>
		/// <param name="packet"></param>
		public virtual void Broadcast(Packet packet)
		{
			lock (_characters)
			{
				foreach (var character in _characters.Values)
					character.Connection.Send(packet);
			}
		}
```

```
		/// <summary>
		/// Broadcasts packet to all characters on map, that are within
		/// visible range of source.
		/// </summary>
		/// <param name="packet">Packet to broadcast.</param>
		/// <param name="source">Reference actor for visible range to broadcast in.</param>
		/// <param name="includeSource">Send to source as well?</param>
		public virtual void Broadcast(Packet packet, IActor source, bool includeSource = true)
		{
			lock (_characters)
			{
				foreach (var character in _characters.Values.Where(a => (includeSource || a != source) && a.Position.InRange2D(source.Position, VisibleRange)))
					character.Connection.Send(packet);
			}
		}
```

Reply:
Well, you're saying,
> it's should be overcomplicated: is it a dummy? Instead of adding to the _characters dictionary add it to the _dummies.

but while duplicating dozens of lines of code is technically simple, it makes future maintenance and understanding the code much more complicated. And in a project that many more people will work on over many more years, that is far from ideal.

Instead, I would argue that an overall less complicated way is to view dummies as that which they are: characters without a connection. One way to handle that could be to check the character type before sending packets. But another, even more opaque way would be to give dummy characters a dummy connection. Simply a connection type that doesn't do anything when `Send` is called. That way it doesn't matter if you ever do send a packet for any reason, as it will effectively be ignored.

I would implement this, but looking at the current buff handlers, it's hard to tell how this would and should affect the existing code, so you would have to do it.